File: frontend/src/services/recommendationService.js
import api from './api'

export const getRecommendationsByProduct = async (
    productId,
    page = 0,
    size = 10,
    sortBy = 'priority'
) => {

    const response = await api.get(
        `api/v1/recommendations/product/${productId}`,
        {
            params: {
                page,
                size,
                sortBy,
            },
        }
    )

    return response.data
}


export const getAllRecommendations = async (
    page = 0,
    size = 10,
    sortBy = 'priority'
) => {

    const response = await api.get(
        'api/v1/recommendations',
        {
            params: {
                page,
                size,
                sortBy,
            },
        }
    )

    return response.data
}


export const getRecommendationById = async (id) => {

    const response = await api.get(
        `api/v1/recommendations/${id}`
    )

    return response.data
}


export const createRecommendation = async (data) => {

    const response = await api.post(
        'api/v1/recommendations',
        data
    )

    return response.data
}


export const updateRecommendation = async (
    id,
    data
) => {

    const response = await api.put(
        `api/v1/recommendations/${id}`,
        data
    )

    return response.data
}


export const deleteRecommendation = async (id) => {

    const response = await api.delete(
        `api/v1/recommendations/${id}`
    )

    return response.data
}


export const getRecommendationsByType = async (
    type,
    page = 0,
    size = 10,
    sortBy = 'priority'
) => {

    const response = await api.get(
        `api/v1/recommendations/type/${type}`,
        {
            params: {
                page,
                size,
                sortBy,
            },
        }
    )

    return response.data
}

File: recommendation-service/src/main/java/com/ecommerce/recommendation/service/RecommendationService.java
package com.ecommerce.recommendation.service;

import com.ecommerce.recommendation.dto.request.RecommendationRequest;
import com.ecommerce.recommendation.dto.response.RecommendationResponse;
import com.ecommerce.recommendation.enums.RecommendationType;
import org.springframework.data.domain.Page;

public interface RecommendationService {

    RecommendationResponse createRecommendation(
            RecommendationRequest request
    );

    RecommendationResponse updateRecommendation(
            Long id,
            RecommendationRequest request
    );

    RecommendationResponse getRecommendationById(
            Long id
    );

    Page<RecommendationResponse> getAllRecommendations(
            int page,
            int size,
            String sortBy
    );

    Page<RecommendationResponse> getRecommendationsByProduct(
            Long productId,
            int page,
            int size,
            String sortBy
    );

    Page<RecommendationResponse> getRecommendationsByType(
            RecommendationType type,
            int page,
            int size,
            String sortBy
    );

    void deleteRecommendation(
            Long id
    );

}